dome.gd

```diff
@@ -131,7 +165,7 @@ func _on_dome_lost_countdown_timer_timeout():
 	$Building/FlowerSprite.show()
 
 func _on_resource_generation_timer_timeout():
-	if resource_type != Globals.ResourceType.RESEARCH:
+	if resource_type != Globals.ResourceType.RESEARCH || researching:
 		generate_resource()
```

Comment:
should this logic be changed? I thought the dome will _not_ produce if researching

Reply:
actually that doesn't make sense to me either

Reply:
Restricts generation of research to be dependent on the researching flag rather than type. Definitely could use improvement, but in testing it behaves as expected.

Reply:
oh because the type IS research, then it should fall back to the flag, got it